docs/configuration/configuration-file.mdx

```diff
@@ -5,224 +5,453 @@ import { StyledContent } from '../../components/styled.elements';
+
+Here's an example of an additional rule applied to `boolean-parameter-prefixes`:
+    
+Property | Type | Description
+---|---|---
+`severity` | `string` | REQUIRED. Possible values: `error`, `warn`, `off`.
+`prefixes` | `[string]` | A list of boolean prefix strings. Each boolean property must have one of these strings as its initial characters. Possible values: `can`, `has`, `is`, `should`. Default values: `has`, `is`.
+
+```yaml
+lint:
+  extends:
+    - recommended
+  rules:
+    boolean-parameter-prefixes:
+      severity: warn
+      prefixes: ["can", "has", "is", "should"]
+    no-unused-components:
+      severity: error
 ```
 
-### lint
+:::success Tip
+If you write custom rules, you can make it so they accept the additional rules option. Just be sure to document them!
+:::
+
+### Different OpenAPI versions and `lint`
+
+Redocly OpenAPI CLI supports OpenAPI Specification versions 2.0, 3.0, and 3.1. Most of the time you will use one of them, but sometimes you may need to configure different rules based on the OpenAPI Specification version your definition is written in. You can do that by using additional configuration sections.
+
+In this example:
+* If version 2 (formerly known as Swagger) then `lint` will prioritize the `oas2Rules` sub-section.
+* If version 3 (OpenAPI 3.x) then `lint` will prioritize the `oas3_0Rules` sub-section.
+* If no version is defined then `lint` will fall back to the `rules` sub-section.
 
-The `lint` section's primary purpose is to define options for the `lint` and `bundle` commands.
-You may also import plugins and extends their configurations.
+```yaml
+lint:
+  extends:
+    - recommended
+  rules:
+    boolean-parameter-prefixes: warn
+    no-unused-components: error
+  oas2Rules:
+    no-unused-components: off
+  oas3_0Rules:
+    boolean-parameter-prefixes: error
+```
 
-Read more about the [lint configuration section](../guides/lint.md).
+See our [built-in rules documentation](../resources/built-in-rules.md) for more information about the built-in rules available in OpenAPI CLI.
 
+### Resolve JSON references ($refs) — or not
 
-### features.mockServer
+The OpenAPI Specification supports `$refs` in several objects to enable content re-use and make longer API definitions easier to deal with. Different tools and implementations of the specification — as well as the human beings who write the actual definitions — may use or require `$refs` in unsupported places.
 
-The API registry supports [the mock server feature](../../api-registry/guides/mock-server-quickstart.md) and allows project owners to enable it for all branches per API version.
-When the mock server is enabled for an API, you can send test requests to it from any API client.
+Starting from version `beta-30`, OpenAPI CLI automatically resolves all `$refs` by default — even in those dark, dusty corners where the OpenAPI Specification does not allow them. This includes primitive values, for example `string`, in description and example fields.
```

Comment:
issue: I don't think this is accurate. I could be wrong, but I think it resolves all `$refs` allowed by OAS and also resolves all scalar fields (not allowed by OAS). There are still other things it does not resolve which are not allowed by OAS. @RomanHotsiy can correct me if I'm wrong here.